package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,197 @@
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Base guesser classes --- :mod:`MDAnalysis.guesser.base`
+================================================================
+
+Derive context-specific guesser classes from the base class in this module.
+
+Classes
+-------
+
+.. autoclass:: GuesserBase
+   :members:
+   :inherited-members:
+
+"""
+from .. import _GUESSERS
+import numpy as np
+from .. import _TOPOLOGY_ATTRS
+import logging
+from typing import Dict
+import copy
+
+logger = logging.getLogger("MDAnalysis.guesser.base")
+
+
+class _GuesserMeta(type):
+    """Internal: guesser classes registration
+
+    When classes which inherit from GuesserBase are *defined*
+    this metaclass makes it known to MDAnalysis.  'context'
+    attribute  are read:
+     - `context` defines the context of the guesser class for example:
+       forcefield specific context  as MartiniGuesser
+       and file specific context as PDBGuesser.
+
+    Eg::
+
+      class FooGuesser(GuesserBase):
+          format = 'foo'
+
+    .. versionadded:: 2.8.0
+    """
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+
+        _GUESSERS[classdict['context'].upper()] = cls
+
+
+class GuesserBase(metaclass=_GuesserMeta):
+    """Base class for context-specific guessers to inherit from
+
+    Parameters
+    ----------
+    universe : Universe, optional
+        Supply a Universe to the Guesser. This then become the source of atom
+        attributes to be used in guessing processes. (this is relevant to how
+        the universe's guess_TopologyAttrs API works.
+        See :meth:`~MDAnalysis.core.universe.Universe.guess_TopologyAttrs`).
+    **kwargs: to pass additional data to the guesser that can be used with
+              different methos.
+
+    .. versionadded:: 2.8.0
+
+    """
+    context = 'base'
+    _guesser_methods: Dict = {}
+
+    def __init__(self, universe=None, **kwargs):
+        self._universe = universe
+        self._kwargs = kwargs
+
+    def update_kwargs(self, **kwargs):
+        self._kwargs.update(kwargs)
+
+    def copy(self):
+        """Return a copy of this Guesser"""
+        kwargs = copy.deepcopy(self._kwargs)
+        new = self.__class__(universe=None, **kwargs)
+        return new
+
+    def is_guessable(self, attr_to_guess):
+        """check if the passed atrribute can be guessed by the guesser class
+
+        Parameters
+        ----------
+        guess: str
+            Atrribute to be guessed then added to the universe
+
+        Returns
+        -------
+        Boolean value
```

Comment:
```suggestion
        bool
```
Very nitpicky, but that makes it pick the right python standard type docs link I believe.